packages/core/src/store/types.ts

```diff
@@ -15,3 +15,51 @@ export type InferStoreRecord<T> = {
 
 /** A record that is of suitable type to be put in DynamoDB. */
 export type StoreRecord = Record<string, string|number>
+
+/** ---------------------- Dealer ---------------------- */
+/**
+ * Custom Dealer Aggregate store record given we need to collapse Deal Metadata into different columns
+ * together with status.
+ */
+export type DealerAggregateStoreRecord = {
+  // PieceCid of an Aggregate `bagy...aggregate`
+  aggregate: string
+  // Encoded block CID of list of pieces in an aggregate.
+  pieces: string
+  // Deal identifier on the Filecoin chain
+  dealMetadataDealId?: number
+  // Deal data type (always 0 for now)
+  dealMetadataDataType?: number
+  // Status of the offered 
+  stat: DealerAggregateStoreRecordStatus
+  // Date when aggregate was added as ISO string
+  insertedAt: string
+  // Date when aggregate information was updated as ISO string
+  updatedAt: string
+}
```

Comment:
could not find a way to rely on the above `InferStoreRecord` to transform this record, given collapsing deal property, as well as mutating `stat`